xbmc/platform/linux/input/LinuxInputDevices.cpp

```diff
@@ -1030,6 +1030,16 @@ void CLinuxInputDevice::GetInfo(int fd)
     m_deviceMaxAxis = std::max(num_rels, num_abs) - 1;
   }
 
+  /* if no better fit and have at least 1 key */
+   if ((m_deviceType == LI_DEVICE_NONE) && (num_keys >= 1))
```

Comment:
no need for brackets around `m_deviceType == LI_DEVICE_NONE` and `num_keys >= 1`